pyright/_utils.py

```diff
@@ -71,6 +74,33 @@ def install_pyright(args: tuple[object, ...], *, quiet: bool | None) -> Path:
     return pkg_dir
 
 
+def _get_configured_pyright_version() -> str:
+    force_version = os.environ.get('PYRIGHT_PYTHON_FORCE_VERSION')
+    if force_version:
+        return force_version
+
+    pylance_version = os.environ.get('PYRIGHT_PYTHON_PYLANCE_VERSION')
+    if pylance_version:
+        return _get_pylance_pyright_version(pylance_version)
+
+    return __pyright_version__
+
+
+def _get_pylance_pyright_version(pylance_version: str) -> str:
+    url = f'https://raw.githubusercontent.com/microsoft/pylance-release/main/releases/{pylance_version}.json'
+
+    try:
+        response = mureq.get(url, timeout=1)
+        response.raise_for_status()
+
+        version = response.json()["pyrightVersion"]
```

Comment:
nit: this is easier to debug if something goes wrong (can't remember if this exact log format syntax is correct)
```suggestion
        data = response.json()
        log.debug('pylance release data', data)

        version = data["pyrightVersion"]
```